### PR TITLE
ncm-systemd: Add path support

### PR DESCRIPTION
**Pull request by DavidFair**
Adds support for [path] directives. This is useful for credential handling, where we can detect the template file changing and trigger a one-shot service.

### CODE REVIEW

**Comment by wpoely86** [comment]
I guess you also want `.path` files? In that case you need to add a couple of extra bits, see #1582 which does this for a `.slice` unit.

File: ncm-systemd/src/main/pan/components/systemd/schema.pan [hunk 1]
```diff
@@ -355,6 +355,24 @@ type ${project.artifactId}_unitfile_config_socket = {
...
+    'PathExists' ? absolute_file_path
+    'PathExistsGlob' ? string
+    'PathChanged' ? absolute_file_path
+    'PathModified' ? absolute_file_path
+    'DirectoryNotEmpty' ? absolute_file_path
+
+    'Unit' ? string
+    'MakeDirectory' ? boolean
+    'DirectoryMode' ? type_octal_mode
+    'TriggerLimitIntervalSec' ? string
```
Comment (jrha) [inline ncm-systemd/src/main/pan/components/systemd/schema.pan:372]:
```suggestion
    'TriggerLimitIntervalSec' ? long(0..)
```

Comment (DavidFair) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:372]: The documentation suggests users put a unit prefix (e.g. 1s or 1us): https://www.freedesktop.org/software/systemd/man/systemd.path.html
Do we have a type which supports this?

**Comment by DavidFair** [comment]
> I guess you also want `.path` files? In that case you need to add a couple of extra bits, see #1582 which does this for a `.slice` unit.

I'm not sure what you mean sorry? I will add a section to the unit tests though based on that PR

**Comment by wpoely86** [comment]
> > I guess you also want `.path` files? In that case you need to add a couple of extra bits, see #1582 which does this for a `.slice` unit.
> 
> I'm not sure what you mean sorry? I will add a section to the unit tests though based on that PR

Can't make a `.path` unit right now with this. You need to a path to the list on line 504 of `schema.pan`

File: ncm-systemd/src/main/pan/components/systemd/schema.pan (hunk 1 above)
Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan]: For simplicity we normally only support the base unit (in this case seconds).

File: ncm-systemd/src/main/pan/components/systemd/schema.pan [hunk 2]
```diff
@@ -355,6 +355,24 @@ type ${project.artifactId}_unitfile_config_socket = {
     true;
 };
 
+@documentation{
+the [Path] section
+https://www.freedesktop.org/software/systemd/man/systemd.path.html
+}
+type ${project.artifactId}_unitfile_config_path = {
+    'PathExists' ? absolute_file_path
+    'PathExistsGlob' ? string
```
[the remark this answers is not in the document]
Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:364]: `absolute_file_path` doesn't care about globs, it just checks that it starts with a leading `/` and doesn't end with one.

[the remark this answers is not in the document]
Comment (jrha) [reply ncm-systemd/src/main/pan/components/systemd/schema.pan:364]: See https://github.com/quattor/template-library-core/blob/master/pan/types.pan#L997-L1016

File: ncm-systemd/src/main/resources/regular/tests/profiles/unitfile.pan [hunk 3]
```diff
@@ -69,6 +69,18 @@ bind "/unitfile" = systemd_unitfile_config[];
         'SocketGroup', 'pipegroup',
         'SocketMode', '660',
         ),
+    'path', dict(
```
Comment (wpoely86) [inline ncm-systemd/src/main/resources/regular/tests/profiles/unitfile.pan:72]: you still need to add the expected output of this section to `ncm-systemd/src/main/resources/regular/tests/regexps/unitfile0/value`

**Comment by jrha** [comment]
@wpoely86 I've rebased this and sorted out the tests, so could you re-review please?

**Review by wpoely86 — APPROVED** [review]

**Comment by DavidFair** [comment]
Thanks for getting this over the line, I've been pretty wrapped up with leave so it unfortunately got left by the wayside

**Pull request merged**